### ISSUE TITLE
Issue 028: Implement MAE pretraining engine with checkpointing

### ISSUE THREAD

**Issue by ozw4** [ozw4/seis-attr-ssl#39]
# Issue 028: Implement MAE pretraining engine with checkpointing

## Goal
Implement the Stage 1 strict MAE pretraining loop.

## Context
The repository currently has dry-run entrypoints. After this issue, `proc/train_mae.py` should be able to run a real small training job when provided with a manifest and valid `.npy` attribute volumes.

F3 must not be used in pretraining.

## Implementation tasks

Create:

```text
src/seis_attr_ssl/training/mae.py
src/seis_attr_ssl/training/checkpoint.py
src/seis_attr_ssl/training/logging.py
```

Implement:

```python
@dataclass
class MaeTrainingState: ...

def train_mae_one_epoch(...): ...
def run_mae_pretraining(config: Mapping[str, object]) -> Path: ...
def save_checkpoint(...): ...
def load_checkpoint(...): ...
```

Training requirements:

- Build `NopimsAttributePretrainDataset` from manifest JSON path in config.
- Use `mae_collate_fn`.
- Instantiate `StrictAttributeSetMAE3D` from config.
- Use `mae_pretraining_loss` from Issue 026.
- Support CPU and CUDA.
- Use AMP when `train.amp=true` and CUDA is available.
- Use AdamW.
- Save checkpoints under `paths.output_root`.
- Save at least:
  - model state dict
  - optimizer state dict
  - epoch
  - config snapshot
  - package version if available
- Print concise epoch metrics to stdout.
- Keep distributed training out of scope.

Config addition needed:

Add manifest path fields to `proc/configs/mvp_mae.yaml`, for example:

```yaml
manifests:
  train: /home/dcuser/data/NOPIMS/manifests/nopims_manifests.json
```

Keep validation conservative and ensure no F3 path appears in pretraining config.

## Acceptance criteria

```bash
ruff check .
pytest -q
python -m compileall -q src proc tests
```

Also add a synthetic smoke test that runs 1 tiny epoch on CPU using temporary `.npy` volumes and verifies:

- training completes
- checkpoint is written
- checkpoint can be loaded
- loss is finite

## Tests to add

Add `tests/test_mae_training_engine.py` covering:

1. one-epoch CPU training on tiny synthetic manifest
2. checkpoint save/load round trip
3. config with F3 path/key is rejected for pretraining
4. AMP flag does not fail on CPU, but does not enable CUDA AMP

## Out of scope

Do not implement:

- dense adaptation
- fine-tuning
- evaluation metrics
- distributed training
